mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -960,7 +962,7 @@ VectorizationResult mlir::rock::getMaxVectorization(
   // warp will issue, if possible, a global_load_dwordx4 instruction
   if (!ignoreDataType) {
     constexpr int64_t maxVectorLenBits = 128;
-    int64_t bwidth = upperType.getElementTypeBitWidth();
+    int64_t bwidth = outputType.getElementTypeBitWidth();
```

Comment:
I think this fixes the issue where getMaxVec returns the final type? (for example fp16 insetad of fp32)? Nice!

Reply:
Yep, went and fixed that here - so maybe we can move the output swizzle creation before align-tiling!